libbeat/monitoring/cloudid.go

```diff
@@ -0,0 +1,119 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package monitoring
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/cloudid"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type cloudConfig struct {
+	Cloud struct {
+		ID   string `config:"id"`
+		Auth string `config:"auth"`
+	} `config:"cloud"`
+}
+
+// ErrCloudCfg represents an error when trying to use monitoring.cloud.* settings.
+type ErrCloudCfg struct {
+	cause error
+}
```

Comment:
Although it's good practice to introduce error types, they sometimes bloat code. Think if you need the error type or some other exported method on the error value (one can use type cases to interfaces to query errors) to be part of your public interface. Typed errors allow you to signal a specific error, allowing users to handle it. But sometimes `errors.Wrap` is enough. It's a fine line between being lazy and allowing devs to handle errors :)

Reply:
@urso I want to make sure I'm understand the distinction correctly. The way I interpret your comment is something like: if a dev might need to do some kind of conditional logic depending on the error returned, then it's better to have a custom error type. But if the need is just to display the error for logging, etc. (i.e. no programmatic inspection of the error is needed) then using `errors.Wrap` is fine. Is this a good way to think about the distinction?

Reply:
Yes. `errors.Wrap` still allows some unpacking in case users want to introspect. The new functions in `errors` (e.g. `Is`) even walk the list of errors in case an error is simply wrapped.
If some package(-hierarchy) is supposed to be consumed as a library you would prefer clear error types or values.

In the end you have to make the call on what makes sense, I just noticed that you started introducing loads of error types lately (some only carry a cause error).